components/pagination/collections/paging_collection.js

```diff
@@ -0,0 +1,258 @@
+ * others. There is support for ascending or descending sort on a particular field, as well as filtering on a field.
+ * While the backend API may use either zero or one indexed page numbers, this collection uniformly exposes a one
+ * indexed interface to make consumption easier for views.
+ *
+ * Subclasses may want to override the following properties:
+ *      - url (string): The base url for the API endpoint.
+ *      - isZeroIndexed (boolean): If true, API calls will use page numbers starting at zero. Defaults to false.
+ *      - perPage (number): Count of elements to fetch for each page.
+ *      - queryParams (object): Query parameters for the API call. Subclasses may add entries as necessary. By default,
+ *          a 'sort_order' field is included to specify the field to sort on. This field may be removed for subclasses
+ *          that do not support sort ordering, or support it in a non-standard way. By default filterField and
+ *          sortDirection do not affect the API calls. It is up to subclasses to add this information to the appropriate
+ *          query string parameters in server_api.
+ */
+;(function (define) {
+    'use strict';
+    define(['jquery', 'backbone.paginator'], function ($, PageableCollection) {
+        var PagingCollection = PageableCollection.extend({
+            initialize: function () {
+                // These must be initialized in the constructor because otherwise all PagingCollections would point
+                // to the same object references for sortableFields and filterableFields.
+                this.sortableFields = {};
+                this.filterableFields = {};
+            },
+
+            isZeroIndexed: false,
+            perPage: 10,
+            mode: 'server',
+
+            isStale: false,
+
+            sortField: '',
+            sortDirection: 'descending',
+            sortableFields: {},
+
+            filterField: '',
+            filterableFields: {},
+
+            searchString: null,
+
+            state: {
+                firstPage: function () { return this.isZeroIndexed ? 0 : 1; },
+                // Specifies the initial page during collection initialization
+                currentPage: function () { return this.isZeroIndexed ? 0 : 1; },
+                perPage: function () { return this.perPage; }
+            },
+
+            queryParams: {
+                page: function () { return this.state.currentPage; },
+                page_size: function () { return this.perPage; },
+                text_search: function () { return this.searchString ? this.searchString : ''; },
+                sort_order: function () { return this.sortField; }
+            },
+
+            parse: function (response) {
+                this.totalCount = response.count;
+                this.state.currentPage = response.current_page;
+                this.totalPages = response.num_pages;
+                this.start = response.start;
```

Comment:
That's what I was hoping... I didn't see it referenced anywhere else in
this PR.

On Tue, Dec 8, 2015 at 2:01 PM, Andy Armstrong <notifications@github.com>
wrote:

> In components/pagination/collections/paging_collection.js
> <https://github.com/edx/edx-ui-toolkit/pull/2#discussion_r46997176>:
>
> > +                currentPage: function () { return this.isZeroIndexed ? 0 : 1; },
> > +                perPage: function () { return this.perPage; }
> > +            },
> > +
> > +            queryParams: {
> > +                page: function () { return this.state.currentPage; },
> > +                page_size: function () { return this.perPage; },
> > +                text_search: function () { return this.searchString ? this.searchString : ''; },
> > +                sort_order: function () { return this.sortField; }
> > +            },
> > +
> > +            parse: function (response) {
> > +                this.totalCount = response.count;
> > +                this.state.currentPage = response.current_page;
> > +                this.totalPages = response.num_pages;
> > +                this.start = response.start;
>
> I think it was the index of the first item in the page. I think it would
> be safe to remove it at this point, as the only fields that the client
> should depend on are:
>
>
> https://openedx.atlassian.net/wiki/display/AC/edX+REST+API+Conventions#edXRESTAPIConventions-7.Pagination
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-ui-toolkit/pull/2/files#r46997176>.
>